js/modules/k6/http/http_request.go

```diff
@@ -44,6 +44,14 @@ import (
 	null "gopkg.in/guregu/null.v3"
 )
 
+type HTTPRequest struct {
+	Method  string
+	URL     *neturl.URL
+	Headers map[string][]string
+	Body    io.Closer
+	Cookies map[string]*HTTPRequestCookie
```

Comment:
The `Cookie` header is allowed to contain multiple cookies with the same name [1] so `Cookies` here needs to be a `map[string][]*HTTPRequestCookie`, or `map[string][]string` would be fine as well in my opinion.

[1] - https://tools.ietf.org/html/rfc6265#section-4.2

Reply:
Here I've just used the same types in the map as are being used for the cookies being put into the request struct: https://github.com/loadimpact/k6/blob/master/js/modules/k6/http/http_request.go#L139 

Does that map need to change as well?

Reply:
No, that map only stores the user specified per-request cookies (from JS, like the first example here: https://docs.k6.io/docs/cookies#section-setting-simple-cookies).

But to get all cookies that are sent with the request (we is what we want to expose through `res.request.cookies`) we also need to look at what's in the cookiejar (from any previous interactions with the server). And that's when the situation can arise where we end up with multiple cookies with the same name.

Merging the per-request cookies set by the user in JS land with any cookies already present in the cookiejar is what https://github.com/loadimpact/k6/blob/master/js/modules/k6/http/http.go#L130 does.